code/modules/power/supermatter/supermatter.dm

```diff
@@ -393,16 +361,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 	var/turf/T = get_turf(src)
 	for(var/mob/M in GLOB.player_list)
-		if(M.get_virtual_z_level() == get_virtual_z_level())
+		if(M.z == z)
 			SEND_SOUND(M, 'sound/magic/charge.ogg')
 			to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
 			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
 	if(combined_gas > MOLE_PENALTY_THRESHOLD)
 		investigate_log("has collapsed into a singularity.", INVESTIGATE_ENGINES)
-		if(T)
+		if(T) //If something fucks up we blow anyhow. This fix is 4 years old and none ever said why it's here. help.
```

Comment:
I'm guessing the reasoning would be to confirm there is a turf, rather than runtiming.

Reply:
there always is a turf though which is why i'm confused as space counts as a turf and you can't remove that